src/SAML2/Provider/HostedIdentityProviderProcessor.php

```diff
@@ -178,16 +181,28 @@ public function onKernelResponse(FilterResponseEvent $event)
             return;
         }
 
-        if($event->getResponse()->isServerError() || $event->getResponse()->isClientError()){
+        if ($event->getResponse()->isServerError() || $event->getResponse()->isClientError()) {
+            return;
+        }
+
+        if ($event->getResponse()->isServerError() || $event->getResponse()->isClientError()) {
             return;
         }
 
-        if ($this->stateHandler->can(SamlStateHandler::TRANSITION_SSO_RESPOND)) {
+        if (
+            $this->stateHandler->get() !== null
+            && $this->stateHandler->get()->getRequest() !== null
+            && $this->stateHandler->can(SamlStateHandler::TRANSITION_SSO_RESPOND)
```

Comment:
This is equivalent, this `stateHandler->can` has been updated to check that the state and the request are not null